### PR TITLE
Microsoft.Data.Sqlite: Make CommandTimeout of 0 mean no timeout

### LINKED ISSUE #14839: Microsoft.Data.Sqlite: Should a CommandTimeout of 0 mean no timeout?
It's recommended, but today, it means fail immediately, don't retry when busy/locked.

### PR DESCRIPTION
**Pull request by bricelam**
Fixes #14839

### CODE REVIEW

**Comment by ErikEJ** [comment]
@roji It is not a part of the EF Core runtime? Just like System.Data.SqlClient is not.

**Comment by roji** [comment]
@ErikEJ but Microsoft.Data.Sqlite is part of EFCore.Runtime.sln (note that the provider has recently been merged into the same repo as EF Core).

**Comment by roji** [comment]
Never mind, I realized it's part of the separate Sqlite solution.

**Comment by bricelam** [comment]
@roji, I'll remove Microsoft.Data.Sqlite from EFCore.Runtime.sln as soon as an issue gets fixed--VS complains without it.

**Comment by bricelam** [comment]
😄 Introduced deadlock in `EFCore.Sqlite.Tests`.

File: test/Microsoft.Data.Sqlite.Tests/SqliteTransactionTest.cs [hunk 1]
```diff
@@ -97,7 +97,7 @@ public void Serialized_disallows_dirty_reads()
                 {
                     connection1.ExecuteNonQuery("UPDATE Data SET Value = 1;");
 
-                    connection2.DefaultTimeout = 0;
+                    connection2.DefaultTimeout = 1;
```
Comment (bricelam) [inline test/Microsoft.Data.Sqlite.Tests/SqliteTransactionTest.cs]: Doh. I swear I searched for that...

**Pull request merged**